### ISSUE TITLE
Client tries to connect before ports are open

### ISSUE THREAD

**Issue by boehmseb** [ocraft/ocraft-s2client#9]
Using the S2Coordinator for launching games, I always run into the issue that the clients try to connect before the starcraft executable is completely launched and has not yet opened its ports.
As a result, an exception is thrown with the message `Connection refused`.

I can confirm that this is indeed a timing issue, pausing for a small amount of time before trying to connect circumvents the issue. It might be worth mentioning that I am using the linux packages available [here](https://github.com/Blizzard/s2client-proto#downloads)

Here is an excerpt of the log and stacktrace:
```2019-04-16 11:28:03,887|INFO |main                            |org.reflections.Reflections                                     |Reflections took 26 ms to scan 1 urls, producing 7 keys and 9 values 
2019-04-16 11:28:03,977|INFO |main                            |com.github.ocraft.s2client.bot.gateway.impl.ControlInterfaceImpl|Waiting for connection...
2019-04-16 11:28:04,172|INFO |main                            |com.github.ocraft.s2client.api.controller.S2Controller          |Launching Starcraft II with configuration: Config(...).
2019-04-16 11:28:04,195|INFO |main                            |com.github.ocraft.s2client.api.controller.S2Controller          |Launched SC2 (../Sommercamp_2019/SC2.4.7.1/StarCraftII/Versions/Base70154/SC2_x64), PID: 9678
2019-04-16 11:28:04,469|INFO |main                            |com.github.ocraft.s2client.api.S2Client                         |Starting: S2Client{connectToIp='127.0.0.1', connectToPort=8167, requestTimeoutInMillis=60000, connectTimeoutInMillis=60000, done=false, traced=true}
2019-04-16 11:28:04,890|ERROR|vert.x-eventloop-thread-0       |com.github.ocraft.s2client.api.vertx.OnConnect                  |OnConnect.onError
io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /127.0.0.1:8167
	at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method) ~[?:?]
	at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:779) ~[?:?]
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:327) ~[netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:340) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:632) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:579) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:496) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:458) [netty-transport-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:897) [netty-common-4.1.30.Final.jar:4.1.30.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.30.Final.jar:4.1.30.Final]
	at java.lang.Thread.run(Thread.java:834) [?:?]
Caused by: java.net.ConnectException: Connection refused```

**Comment by ocraft** [comment]
Thanks for the info. Automatic detection of configuration and running of the game was tested against normal/full distribution of Starcraft 2, linux packages are better used as standalone/already running server to which you connect as a client:
```java       
S2Coordinator s2Coordinator = S2Coordinator.setup()
                .loadSettings(new String[]{"-m", "Lava Flow"})
                .setParticipants(
                        S2Coordinator.createParticipant(Race.TERRAN, new S2Agent() {
                        }),
                        S2Coordinator.createComputer(Race.ZERG, Difficulty.MEDIUM))
                .connect("127.0.0.1", 5000);
```

Nevertheles it should work with linux packages also, I'll try to fix that with the next release.

**Comment by ocraft** [comment]
Fixed in -> Linux fixes #13
